package/MDAnalysis/analysis/rms.py

```diff
@@ -398,125 +416,100 @@ def __init__(self, traj, reference=None, select='all',
+
+        # reference centre of mass system, no longer 1 based indices
+        current_frame = self.reference.trajectory.ts.frame
 
-        # reference centre of mass system
-        current_frame = self.reference.trajectory.ts.frame - 1
         try:
             # Move to the ref_frame
             # (coordinates MUST be stored in case the ref traj is advanced
             # elsewhere or if ref == mobile universe)
-            self.reference.trajectory[ref_frame]
-            ref_com = self.ref_atoms.center_of_mass()
+            self.reference.trajectory[self.ref_frame]
+            self._ref_com = self.ref_atoms.center_of_mass()
             # makes a copy
-            ref_coordinates = self.ref_atoms.positions - ref_com
-            if self.groupselections_atoms:
-                groupselections_ref_coords_64 = [
-                    self.reference.select_atoms(*s['reference']).positions.astype(np.float64) for s in
+            self._ref_coordinates = self.ref_atoms.positions - self._ref_com
+            if self._groupselections_atoms:
+                self._groupselections_ref_coords_64 = [(self.reference.
+                    select_atoms(*s['reference']).
+                    positions.astype(np.float64)) for s in
                     self.groupselections]
         finally:
             # Move back to the original frame
             self.reference.trajectory[current_frame]
-        ref_coordinates_64 = ref_coordinates.astype(np.float64)
 
-        # allocate the array for selection atom coords
-        traj_coordinates = traj_atoms.positions.copy()
+        self._ref_coordinates_64 = self._ref_coordinates.astype(np.float64)
 
-        if self.groupselections_atoms:
+        if self._groupselections_atoms:
             # Only carry out a rotation if we want to calculate secondary
             # RMSDs.
             # R: rotation matrix that aligns r-r_com, x~-x~com
             #    (x~: selected coordinates, x: all coordinates)
             # Final transformed traj coordinates: x' = (x-x~_com)*R + ref_com
-            rot = np.zeros(9, dtype=np.float64)  # allocate space
-            R = np.matrix(rot.reshape(3, 3))
+            self._rot = np.zeros(9, dtype=np.float64)  # allocate space
+            self._R = np.matrix(self._rot.reshape(3, 3))
+        else:
+            self._rot = None
+
+        self.rmsd = np.zeros((self.n_frames,
+                              3 + len(self._groupselections_atoms)))
+
+        self._pm.format = ("RMSD %(rmsd)5.2f A at frame "
+                           "%(step)5d/%(numsteps)d  [%(percentage)5.1f%%]\r")
+
+    def _single_frame(self):
+        mobile_com = self.mobile_atoms.center_of_mass().astype(np.float64)
+        mobile_coordinates = self.mobile_atoms.positions - mobile_com
+        self.rmsd[self._frame_index, :2] = self._ts.frame, self._trajectory.time
+
+        if self._groupselections_atoms:
+            # 1) superposition structures. No longer need to transpose coordinates
```

Comment:
I know that this is not your comment but while you're at it could you please rewrite it as
```python
# 1) superimpose structures: MDAnalysis qcprot needs Nx3 coordinate array with float64
#    datatype (float32 leads to errors up to 1e-3 in RMSD).
#    Note that R is defined in such a way that it acts **to the left** so that we can easily
#    use broadcasting and save one expensive numpy transposition.
```
```